src/app/utils/dataLoaderUtil.js

```diff
@@ -0,0 +1,109 @@
+import Actions from '@Actions';
+import { ajaxCall } from '@utils';
+import appConfig from '@appConfig';
+import Store from '@Store';
+
+const pathInstructions = [
+  {
+    expression: /\/research\/collections\/shared-collection-catalog\/bib\/([cp]?b\d*)/,
+    pathType: 'bib',
+  },
+  {
+    expression: /\/research\/collections\/shared-collection-catalog\/search\?(.*)/,
+    pathType: 'search',
+  },
+  {
+    expression: /\/research\/collections\/shared-collection-catalog\/hold\/request\/([^/]*)/,
+    pathType: 'holdRequest',
+  },
+];
+
+const routesGenerator = (location, next) => ({
+  bib: {
+    apiRoute: matchData => `${next ? 'http://localhost:3001' : ''}${appConfig.baseUrl}/api/bib?bibId=${matchData[1]}`,
+    actions: [Actions.updateBib],
+    errorMessage: 'Error attempting to make an ajax request to fetch a bib record from ResultsList',
+  },
```

Comment:
`Actions.updateBib` is the right handler for client-side as it calls `setState`. Is that true for server-side? Didn't the old code overwrite `res.locals.data.Store` on the server-side so that the component would have the data available on-render? Does what `Actions.updateBib` do have the same effect?

Reply:
Yes. In order to bring the server side and client side into alignment, I've changed `server.js` to bootstrap from the `Store` instead of `res.locals.data.Store`